autoload/lsp/omni.vim

```diff
@@ -201,7 +207,35 @@ function! lsp#omni#get_vim_completion_item(item, ...) abort
         endif
     endif
 
+    if exists('l:snippet')
+        let l:completion['user_data'] = string([l:trigger, l:snippet])
+    endif
+
     return l:completion
 endfunction
 
+function! s:expand_snippet(timer) abort
+    call feedkeys("\<C-r>=UltiSnips#Anon(\"" . s:snippet . "\", \"" . s:trigger . "\", '', 'i')\<CR>")
+endfunction
+
+function! s:handle_snippet(item) abort
+    if !has_key(a:item, 'user_data')
+        return
+    endif
+
+    execute 'let l:user_data = ' . a:item['user_data']
+
+    let s:trigger = l:user_data[0]
+    let s:snippet = l:user_data[1]
+
+    call timer_start(0, function('s:expand_snippet'))
```

Comment:
Curious why do we need a timer here?

Reply:
I'm not 100% sure why, but without a timer the tabstops in nested snippets do not work:
![without-timer](https://user-images.githubusercontent.com/10748726/53116104-1f445480-3548-11e9-8e93-51ef47a12ce4.gif)

It seems only the innermost snippet's tabstops are preserved.